unit/util/irep_sharing.cpp

```diff
@@ -7,7 +7,7 @@
 
 #ifdef SHARING
 
-SCENARIO("irept_sharing_trade_offs", "[!mayfail][core][utils][irept]")
+SCENARIO("irept_sharing_trade_offs", "[!shouldfail][core][utils][irept]")
```

Comment:
I'm not sure I agree with this one: it should actually not fail, it's just a currently known bug in the implementation... (the same is true of the other change in this file). I'd suggest @NathanJPhillips comment on this.

Reply:
(Nathan is AL atm, though this isn't critical so can wait if you don't agree with below)

I personally think if it starts _not_ failing, we should turn it on (i.e. remove the flag). This current flag allows it to fluctuate between working and not working without anyone ever noticing. By changing it to `!shouldfail`, if someone fixes it, the tests will fail and they will know to turn them on (and not accidentally write a bunch more!)